### DISCUSSION TITLE
May i ask why we place the loadbalancer in a different vcn with the main vcn?

### DISCUSSION THREAD

**Discussion by JinhuW** [oracle-quickstart/oci-cloudnative#333] [Q&A] [answered]
Hi,
May I ask why we design to place the load balancer in a different VCN with the main VCN?
https://github.com/oracle-quickstart/oci-cloudnative/blob/master/deploy/basic/terraform/network.tf#L13
Is it the best practice for it?

Thanks

**Comment by junior** [comment, marked as answer]
If you check the code, you will see that the secondary vcn is optional, and the behavior is driven by the `create_secondary_vcn` variable.

The standard architecture when deploying does not create the secondary VCN. 
![00-Topology-v1 2 0](https://user-images.githubusercontent.com/551007/201970355-f9a87851-e241-4a75-bef5-ee0e44c35e4a.svg)
Reference: https://github.com/oracle-quickstart/oci-cloudnative/blob/master/deploy/basic/README.md

---
The option that creates a secondary VCN is when you use [Security Zones](https://docs.oracle.com/en-us/iaas/security-zone/using/security-zones.htm), which does not allow internet ingress. Hear the architecture using [Security Zones](https://docs.oracle.com/en-us/iaas/security-zone/using/security-zones.htm).
![00-Topology-v1 2 0 MSZ](https://user-images.githubusercontent.com/551007/201970507-40e733c3-7800-476e-aad9-5cb496aca707.svg)
Reference: https://github.com/oracle-quickstart/oci-cloudnative/blob/master/deploy/basic/README_MSZ.md

**Reply by JinhuW** [reply]
I appreciate it!! Thanks for your detailed explanation.
